runhouse/servers/cluster_servlet.py

```diff
@@ -217,3 +228,82 @@ async def aclear_all_references_to_env_servlet_name(self, env_servlet_name: str)
         for key in deleted_keys:
             self._key_to_env_servlet_name.pop(key)
         return deleted_keys
+
+    ##############################################
+    # Cluster status functions
+    ##############################################
+
+    async def asend_status_info_to_den(self):
+        while True:
+            logger.info("Sending cluster status to Den")
+            interval_size = (await self.aget_cluster_config()).get(
+                "den_status_ping_interval"
+            )
+            try:
+                status = await self.astatus()
+                status_data = {
+                    "status": "running",
+                    "resource_type": status.get("cluster_config").get("resource_type"),
+                    "data": status,
+                }
+                cluster_uri = rns_client.format_rns_address(
+                    status.get("cluster_config").get("name")
+                )
+                api_server_url = status.get("cluster_config").get(
+                    "api_server_url", rns_client.api_server_url
+                )
+                post_status_data_resp = requests.post(
+                    f"{api_server_url}/resource/{cluster_uri}/cluster/status",
+                    data=json.dumps(status_data),
+                    headers=rns_client.request_headers(),
+                )
+                if post_status_data_resp.status_code != 200:
+                    logger.error(
+                        f"({post_status_data_resp.status_code}) Failed to send cluster status to Den: {post_status_data_resp.text}"
+                    )
+                else:
+                    logger.info(
+                        f"Successfully updated cluster status in Den. Next ping to den in {round(DEFAULT_STATUS_CHECK_INTERVAL / 60, 2)} minutes."
+                    )
+            except Exception as e:
+                logger.error(
+                    f"Failed to post cluster status to den: {e}. Please check cluster logs for more info."
+                )
+                logger.info(
+                    "Temporarily increasing the interval between two status pings to den. Next ping will be in an hour."
+                )
+                await asyncio.sleep(3600)
+            finally:
+
+                await asyncio.sleep(interval_size)
+
+    def send_status_info_to_den_sync(self):
+        asyncio.run(self.asend_status_info_to_den())
+
+    def schedule_post_status(self):
```

Comment:
I think we can probably use an asyncio periodic here, instead of a thread. I'd start with that and switch to a thread if needed. The cluster servlet isn't doing expensive work so I'm not too worried about it.

More on this: https://stackoverflow.com/questions/37512182/how-can-i-periodically-execute-a-function-with-asyncio

Also an old PR of mine that does this, you can follow the pattern: https://github.com/run-house/runhouse/pull/701